### PR TITLE
Signatures using the Core SDK

### PR DESCRIPTION
**Pull request by germartinez**
## Context:
The SDK team is updating the documentation to explain how to create signatures using the Safe{Core} SDK:
- Utilities to generate signatures using the SDK
- Signature collection to execute Transactions or to sign Messages
- Signature collection using the SDK and Safe Services
- on-chain vs off-chain messages

## Note:
This PR cherry picks all the commits from https://github.com/safe-global/safe-docs/pull/313 as that PR was created from the wrong base branch. This one is using `development` after `nextra` was squash merged.

### CODE REVIEW

File: pages/sdk-protocol-kit/guides/safe-signatures.md [hunk 1]
```diff
@@ -0,0 +1,853 @@
+# Safe Signatures
+
+Understanding and generating signatures can be challenging. In the **Safe{Core} SDK**, we provide a set of utilities that can assist in using signatures with Safe. In this article, we will explain how signatures work and how to generate them using the `@safe-global/protocol-kit` package.
+
+## Setting up the example Safe Account
+
+Your Safe Account can be configured with various thresholds and different owner types. An owner can be any Ethereum address, such as:
```
Comment (germartinez) [inline pages/sdk-protocol-kit/guides/safe-signatures.md:7]:
```suggestion
Your Safe account can be configured with different values for the threshold and different types of owner. An owner can be any Ethereum address, such as:
```

File: pages/sdk-protocol-kit/guides/safe-signatures.md [hunk 2]
```diff
@@ -0,0 +1,853 @@
...
+
+- External Owned Account (EOA)
+- An smart contract that implements [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) for signature validation, such as Safe Account
```
Comment (germartinez) [inline pages/sdk-protocol-kit/guides/safe-signatures.md:10]:
```suggestion
- An smart contract that implements [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) for signature validation, such as another Safe.
```

File: pages/sdk-protocol-kit/guides/safe-signatures.md [hunk 3]
```diff
@@ -0,0 +1,853 @@
...
+
+When the owner is an EOA, an ECDSA signature is generated. When the signer is a smart account, the exact signature algorithm depends on the signer itself.
+
+In this article, we will use the following Safe Account setup. We've five different Ethereum addresses that can act as signers, namely _owner1_ to _owner5_.
```
Comment (germartinez) [inline pages/sdk-protocol-kit/guides/safe-signatures.md:14]:
```suggestion
In this article, we will use the following Safe account setup. We've five different Ethereum addresses that can act as signers, namely `owner1` to `owner5`.
```

File: pages/sdk-protocol-kit/guides/safe-signatures.md [hunk 4]
```diff
@@ -0,0 +1,853 @@
...
+    - `owner3`
+  - `signerSafe2_3`: 2/3 Safe Account as another child owner
+    - `owner4`
+    - `owner5`
+
+All the owners are Ethereum addresses. Here are the addresses for this example:
+
+| Who                | Address                                    |
+| ------------------ | ------------------------------------------ |
+| **safeAddress3_4** | 0xb3b3862D8e38a1E965eb350B09f2167B2371D652 |
```
Comment (germartinez) [inline pages/sdk-protocol-kit/guides/safe-signatures.md:29]:
```suggestion
| **safe3_4** | 0xb3b3862D8e38a1E965eb350B09f2167B2371D652 |
```

File: pages/sdk-protocol-kit/guides/safe-signatures.md [hunk 5]
```diff
@@ -0,0 +1,853 @@
...
+
+We can sign transactions using the `protocol-kit` by creating an instance of the `Safe` class:
+
+```typescript
+import Safe from '@safe-global/protocol-kit';
+
+// You can use any compatible adapter, such as Web3Adapter or EthersAdapter
+const protocolKit = await Safe.create({
+  ethAdapter: ethAdapter1,
+  safeAddress: safeAddress3_4,
```
Comment (germartinez) [inline pages/sdk-protocol-kit/guides/safe-signatures.md:50]:
```suggestion
  safeAddress: safe3_4,
```

**Comment by germartinez** [comment]
Acording to the code style in the docs all semicolons should be removed.

**Comment by yagopv** [comment]
@germartinez all the suggestions and comments were addressed

**Review by yagopv — APPROVED** [review]

File: pages/sdk-protocol-kit/guides/signatures/transactions.mdx [hunk 6]
```diff
@@ -0,0 +1,476 @@
...
+// Add signature from the owner safe2_3
+await apiKit.confirmTransaction(
+  safeTransactionHash,
+  buildSignatureBytes([signerSafeSig2_3])
+)
+```
+
+At this point, the transaction stored in the Safe Transaction Service contains all the required signatures from the owners of the Safe.
+
+The `getTransaction` method returns the transaction with the `confirmations` property, used to check all the signatures that were added.
```
Comment (tanay1337) [inline pages/sdk-protocol-kit/guides/signatures/transactions.mdx:420]:
```suggestion
The `getTransaction` method returns the transaction with the `confirmations` property to check all the added signatures.
```

File: pages/sdk-protocol-kit/guides/signatures/transactions.mdx [hunk 7]
```diff
@@ -0,0 +1,476 @@
...
+
+[Safe\{Wallet\}](https://app.safe.global) exposes to its users the list of pending transactions.
+
+```
+https://app.safe.global/transactions/queue?safe=<NETWORK_PREFIX>:<SAFE_ADDRESS>
+```
+
+## Execute the transaction
+
+Connect the Safe and the adapter of an owner to the Protocol Kit. Ensure there are enough funds available in the owner's account to execute the transaction and cover the gas costs. Once the Protocol Kit is initialized, the `executeTransaction` method receives the transaction with the required signatures and executes it.
```
Comment (tanay1337) [inline pages/sdk-protocol-kit/guides/signatures/transactions.mdx:438]:
```suggestion
Connect the Safe and the adapter of an owner to the Protocol Kit. Ensure enough funds are available in the owner's account to execute the transaction and cover the gas costs. Once the Protocol Kit is initialized, the `executeTransaction` method receives and executes the transaction with the required signatures.
```

**Review by tanay1337 — CHANGES_REQUESTED** [review]

**Review by tanay1337 — APPROVED** [review]

**Pull request merged**